psp/exp_configs/island_exc.py

```diff
@@ -0,0 +1,101 @@
+"""Main config for the "island" use-case."""
+
+import datetime as dt
+
+import xarray as xr
+
+from psp.data_sources.nwp import NwpDataSource
+from psp.data_sources.pv import NetcdfPvDataSource, PvDataSource
+from psp.dataset import PvSplits, auto_date_split, split_pvs
+from psp.exp_configs.base import ExpConfigBase
+from psp.models.base import PvSiteModel, PvSiteModelConfig
+from psp.models.recent_history import RecentHistoryModel
+from psp.models.regressors.decision_trees import SklearnRegressor
+from psp.typings import Horizons
+
+PV_TARGET_DATA_PATH = "/mnt/storage_b/data/ocf/solar_pv_nowcasting/clients/island/pv_hourly_v6.nc"
+NWP_PATH = "/mnt/storage_b/data/ocf/solar_pv_nowcasting/clients/island/5y_nwp_3_to_48.zarr"
+EXC_PATH = (
+    "/mnt/storage_b/data/ocf/solar_pv_nowcasting/experimental/Excarta/"
+    "sr_UK_Malta_full/zarr_format/r4/Malta/merged_full/malta_excarta_hour_shift.zarr"
+)
+
+
+def _get_capacity(data: xr.Dataset) -> float:
+    # Use the "capacity" data variable.
+    # There is one capacity per timestamp, let's take the first one arbitrarily; hopefully there
+    # is not a big variation of capacity in our history of a few days!
+    return float(data["capacity"].values[0])
+
+
+class ExpConfig(ExpConfigBase):
+    def get_pv_data_source(self):
+        return NetcdfPvDataSource(
+            PV_TARGET_DATA_PATH,
+        )
+
+    def get_data_source_kwargs(self):
+        return dict(
+            pv_data_source=NetcdfPvDataSource(
+                PV_TARGET_DATA_PATH,
+                lag_minutes=5 * 24 * 60,
+            ),
+            nwp_data_source=NwpDataSource(
+                NWP_PATH,
+                coord_system=4326,
+                x_dim_name="latitude",
+                y_dim_name="longitude",
+                x_is_ascending=False,
+            ),
+            # so to change this just use the existing NWP data source
+            excarta_data_source=NwpDataSource(
+                EXC_PATH,
+                coord_system=4326,
+                x_dim_name="x",
+                y_dim_name="y",
+                time_dim_name="init_time",
+                x_is_ascending=True,
+                single_point=False,
+            ),
+        )
+
+    def get_model_config(self):
+        return PvSiteModelConfig(
+            horizons=Horizons(
+                duration=60,
+                num_horizons=48,
+            )
+        )
+
+    def get_model(self, **kwargs) -> PvSiteModel:
+        return RecentHistoryModel(
+            self.get_model_config(),
+            **self.get_data_source_kwargs(),
+            regressor=SklearnRegressor(
+                num_train_samples=1000,
+                normalize_targets=True,
+            ),
+            use_nwp=True,
```

Comment:
A potentially cleaner solution to this could be to change `use_nwp` to either be `False` or use a list of providers to use. This way, we don't add a `use_<provider>` every time we add a new NWP to use, and instead can just add/remove a provider in a list.